### PR TITLE
Custom gpu driver

### PR DESCRIPTION
**Pull request by innovation2013**
There are currently some issues where things other than the default GPU driver are not working properly.

### CODE REVIEW

**Comment by GPUCode** [comment]
Please run clang format, thanks

**Comment by GPUCode** [comment]
Also move dynamic library to common, it doesn't belong in renderer_vulkan

File: src/common/dynamic_library.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <atomic>
+#include <string>
+
+namespace Vulkan {
```
Comment (GPUCode) [inline src/common/dynamic_library.h:5]:
```suggestion
namespace Common {
```

File: src/common/dynamic_library.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+
+#include <dlfcn.h>
+
+#endif
+
+namespace Vulkan {
```
Comment (GPUCode) [inline src/common/dynamic_library.cpp:14]:
```suggestion
namespace Common {
```

File: src/common/dynamic_library.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+void* DynamicLibrary::GetSymbolAddress(const char* name) const {
+#ifdef _WIN32
+    return reinterpret_cast<void*>(GetProcAddress(reinterpret_cast<HMODULE>(m_handle), name));
+#else
+    return reinterpret_cast<void*>(dlsym(m_handle, name));
+#endif
+}
+} // namespace Vulkan
```
Comment (GPUCode) [inline src/common/dynamic_library.cpp:80]:
```suggestion
} // namespace Common
```

File: src/common/dynamic_library.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        *ptr = reinterpret_cast<T>(GetSymbolAddress(name));
+        return *ptr != nullptr;
+    }
+
+private:
+    // Platform-dependent data type representing a dynamic library handle.
+    void* m_handle = nullptr;
+};
+
+} // namespace Vulkan
```
Comment (GPUCode) [inline src/common/dynamic_library.h:51]:
```suggestion
} // namespace Common
```

File: src/common/dynamic_library.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
+#pragma once
```
Comment (GPUCode) [inline src/common/dynamic_library.h:1]: Both .h/.cpp files need the license header

File: src/common/dynamic_library.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ * Provides a platform-independent interface for loading a dynamic library and retrieving symbols.
+ * The interface maintains an internal reference count to allow one handle to be shared between
+ * multiple users.
+ */
+class DynamicLibrary final {
+public:
+    // Default constructor, does not load a library.
+
+    // Automatically loads the specified library. Call IsOpen() to check validity before use.
+    DynamicLibrary(std::string const& filename = {});
```
Comment (GPUCode) [inline src/common/dynamic_library.h:16]:
```suggestion
    DynamicLibrary(std::string_view filename = {});
```

File: src/video_core/renderer_vulkan/vk_instance.cpp [hunk 7]
```diff
@@ -213,7 +213,7 @@ std::vector<std::string> GetSupportedExtensions(vk::PhysicalDevice physical) {
 Instance::Instance(bool validation, bool dump_command_buffers)
     : enable_validation{validation}, dump_command_buffers{dump_command_buffers} {
     auto vkGetInstanceProcAddr =
-        GetVulkanLoader().getProcAddress<PFN_vkGetInstanceProcAddr>("vkGetInstanceProcAddr");
+        (PFN_vkGetInstanceProcAddr)GetVulkanLoader().GetSymbolAddress("vkGetInstanceProcAddr");
```
Comment (GPUCode) [inline src/video_core/renderer_vulkan/vk_instance.cpp:216]: Can this use GetSymbol?

File: src/android/app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -164,6 +164,7 @@
     <string name="preferences_graphics">Graphics</string>
     <string name="preferences_audio">Audio</string>
     <string name="preferences_debug">Debug</string>
+    <string name="preferences_gpudriver">Gpu Driver</string>
```
Comment (sakuramipha) [inline src/android/app/src/main/res/values/strings.xml:167]:
```suggestion
    <string name="preferences_gpudriver">Install GPU driver</string>
```

**Comment by GPUCode** [comment]
Could you test out this Turnip driver: https://github.com/K11MCH1/AdrenoToolsDrivers/releases/tag/v23.2.0_A7XX
It was reported that it works okay

**Comment by innovation2013** [comment]
All right,I tested this driver and it seems to be working well

**Review by GPUCode — APPROVED** [review]

**Pull request merged**